### ISSUE TITLE
convert from Scala Option to Optional in javadsl packages

### ISSUE THREAD

**Issue by rkuhn** [akka/akka#19441]
using scala-java8-compat library

**Comment by hepin1989** [comment]
+1

**Comment by rkuhn** [comment]
We’re doing these changes for the still-experimental Streams & HTTP only, to see how difficult and breaking it is. We’ll work on the rest of Akka afterwards.

**Comment by ktoso** [comment]
This should be closed when step 2 (applying this in all HTTP classes is completed)

**Comment by 2beaucoup** [comment]
Will this be done for functions as well?

**Comment by ktoso** [comment]
Functions is a trickier question I think... since java has Function and BiFunction, but from 3params onwards you're on your own... Perhaps it's more consistent to use our Function types everywhere // cc @rkuhn @drewhk

**Comment by drewhk** [comment]
> This should be closed when step 2 (applying this in all HTTP classes) is completed

@ktoso wdym? I though I changed all relevant HTTP classes.

**Comment by rkuhn** [comment]
There are a few left in
~~~
akka-http-core/src/main/java/akka/http/javadsl/HttpsContext.java
akka-http-core/src/main/scala/akka/http/javadsl/Http.scala
akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala
akka-http/src/main/scala/akka/http/javadsl/server/values/*
~~~

**Comment by 2m** [comment]
Fixed in #19441
